validation-test/Reflection/reflect_UnsafeContinuation.swift

```diff
@@ -0,0 +1,57 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -lswiftSwiftReflectionTest %s -o %t/reflect_UnsafeContinuation
+// RUN: %target-codesign %t/reflect_UnsafeContinuation
+
+// RUN: %target-run %target-swift-reflection-test %t/reflect_UnsafeContinuation | %FileCheck %s --check-prefix=CHECK-%target-ptrsize
+
+// REQUIRES: reflection_test_support
+// REQUIRES: executable_test
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: ASAN
+
+import SwiftReflectionTest
+
+struct MyValue {
+    let u: UInt
+}
+
+struct MyError: Error {
+    let i: Int
+}
+
+@available(SwiftStdlib 5.1, *)
+class MyClass {
+    let cont: UnsafeContinuation<MyValue, any Error>
+    
+    init(cont: UnsafeContinuation<MyValue, any Error>) {
+        self.cont = cont
+    }
+}
+
+if #available(SwiftStdlib 5.1, *) {
+  _ = try await withUnsafeThrowingContinuation { unsafeContinuation in
+    let myClass = MyClass(cont: unsafeContinuation)
+    reflect(object: myClass)
+    unsafeContinuation.resume(returning: MyValue(u: 1))
+  }
+}
+
+// CHECK: Reflecting an object.
+// CHECK-NEXT: Instance pointer in child address space: 0x{{[0-9a-fA-F]+}}
+// CHECK-NEXT: Type reference:
+// CHECK-NEXT: (class reflect_UnsafeContinuation.MyClass)
+
+// CHECK-64: Type info:
+// CHECK-64-NEXT: (class_instance size=24 alignment=8 stride=24 num_extra_inhabitants=0 bitwise_takable=1
+// CHECK-64-NEXT:   (field name=cont offset=16
+// CHECK-64-NEXT:     (struct size=8 alignment=8 stride=8 num_extra_inhabitants=2147483647 bitwise_takable=1
+// CHECK-64-NEXT:       (field name=context offset=0
+// CHECK-64-NEXT:         (reference kind=strong refcounting=native)))))
```

Comment:
I'd love for someone who is familiar with UnsafeContinuation to double-check that this layout looks reasonable.

Reply:
I'm not as unfamiliar with the low-level implementation as I probably should be, but from a quick stroll through some relevant compiler sources, it looks like an unsafe continuation is represented as a pointer to the corresponding task. It's not refcounted, exactly, as it borrows a +1 from the code creating it, and transfers that +1 to the task when the continuation is resumed (one reason of many why they must be resumed exactly once!) and isn't retained/released in the meantime, but it is effectively a strong reference during its lifetime.

Reply:
Thank you for taking a close look.